src/frontend/src/icp/components/core/IcLoaderWallets.svelte

```diff
@@ -0,0 +1,71 @@
+<script lang="ts">
+	import { onDestroy } from 'svelte';
+	import { initIcrcWalletWorker } from '$icp/services/worker.icrc-wallet.services';
+	import { initIcpWalletWorker } from '$icp/services/worker.icp-wallet.services';
+	import type { WalletWorker } from '$icp/types/ic-listener';
+	import { enabledIcNetworkTokens } from '$lib/derived/network-tokens.derived';
+	import type { TokenId } from '$lib/types/token';
+	import { debounce, nonNullish } from '@dfinity/utils';
+	import { toastsError } from '$lib/stores/toasts.store';
+	import { i18n } from '$lib/stores/i18n.store';
+	import { replacePlaceholders } from '$lib/utils/i18n.utils';
+
+	let workers: Map<TokenId, WalletWorker> = new Map<TokenId, WalletWorker>();
+
+	const manageWorkers = async () => {
+		const unusedWorkers: TokenId[] = Array.from(workers.keys()).filter(
+			(tokenId) => !$enabledIcNetworkTokens.some(({ id: icTokenId }) => icTokenId === tokenId)
+		);
+
+		unusedWorkers.forEach((tokenId) => {
+			const worker = workers.get(tokenId);
+
+			if (nonNullish(worker)) {
+				worker.stop();
+				workers.delete(tokenId);
+			}
+		});
+
+		await Promise.all(
+			$enabledIcNetworkTokens.map(async (token) => {
+				if (!workers.has(token.id)) {
+					try {
+						const worker = await (token.standard === 'icrc'
+							? initIcrcWalletWorker(token)
+							: initIcpWalletWorker());
+
+						worker.stop();
+						worker.start();
+
+						workers.set(token.id, worker);
+					} catch (err: unknown) {
+						// We show the error to the user, but we keep loading the rest of the workers
+						toastsError({
+							msg: {
+								text: replacePlaceholders($i18n.tokens.error.loading_ic_wallet, {
+									$symbol: token.symbol
+								})
+							},
+							err
+						});
+					}
+				}
+			})
+		);
+	};
+
+	const debounceManageWorkers = debounce(manageWorkers, 500);
```

Comment:
Why debouncing? Why 500ms? Or generally speaking, there isn't a way to not require debounce? I always see debounce as a last resort.

Reply:
No particular reason to be honest. We are doing the same for the balances of ERC20 tokens. so I thought we would want to keep the same pattern. But more than happy to remove it, I see it in the same way you do

Reply:
We have of course to double check that it isn't called multiple times but if it isn't, then the debounce is here for no reason and should be removed.

Reply:
During the session it is not called several times, but at loading time it may happen that it is called a few times consecutively: the store `enabledIcNetworkTokens` (that we use for this) is purely a filter of enabled IC tokens done on store `networkTokens`, that is a filter of all tokens in the selected network.

That means that the function `manageWorkers` is triggered even by other things like ETH/ERC20(/BTC) tokens at loading, when everything is being created.

I checked and the same happens in the ERC20 balance loaders.

I can try to improve the filter-stores in a separate PR

Reply:
In the equivalent store for ERC20 tokens `enabledErc20NetworkTokens` there is this comment

```typescript
/**
 * It isn't performant to post filter again the Erc20 tokens that are enabled for the specific selected network or no network selected but, it's code wise convenient to avoid duplication of logic.
 */
```

Reply:
Let's add a comment and TODO. Generally speaking, we should avoid adding debounce when it is not absolutely required.  This can lead to some unexpected errors. So if there is a chance to not use a debounce we should try but, let's postpone.

Reply:
done